design-documents/module-split.md

```diff
@@ -0,0 +1,81 @@
+# Module split
+
+## About
+
+This document clarifies the following two questions of the original [service isolation proposal](service-isolation.md)
+1. What new modules should be called
+2. What to do with module namespaces, keep old or change so they match new module name
+
+## Design
+
+1. Module naming
+
+    The following is proposed naming for modules:
+    API (service contracts)
+    * CatalogApi
+    
+    The following is proposed naming for modules:
+    Admin Ui and admin related business logic (blocks, controllers, static resources, models and resources potentially)
+    * CatalogAdmin
+    * CatalogAdminUi
+    * CatalogBackoffice
+    
+    Storefront Ui and storefront related business logic (blocks, controllers, static resources, models and resources potentially)
+    * CatalogStorefrontUi
+    * CatalogStorefront
+    * CatalogStore
+    
+    Cron (models and resources potentially)
+    * CatalogCron
+    
+    CLI (models and resources potentially)
+    * CatalogCli
+    
+    WebApi (models and resources potentially)
+    * CatalogWebApi
+    
+    Common code (business logic used by multiple areas, models, resource modelss)
+    * CatalogCommon
+    * CatalogShared
```

Comment:
Not sure about `Common` and `Shared`, it assumes that the code under these namespaces knows about its usage in the different areas. I propose to combine these namespaces into something like `Domain`.

Reply:
Added to the list.

Reply:
CatalogBL (business logic)
Catalog DL (domain logic)